autorosters/autorosters_main.py

```diff
@@ -0,0 +1,270 @@
+from mwrogue.esports_client import EsportsClient
+from mwrogue.auth_credentials import AuthCredentials
+import math
+
+
+class AutoRostersRunner(object):
+    def __init__(self, site: EsportsClient, overview_page, tabs):
+        self.site = site
+        self.overview_page = self.site.cache.get_target(overview_page)
+        self.tabs = tabs
+        self.match_data = {}
+        self.alt_teamnames = {}
+        self.rosters_data = {}
+        self.PAGE_HEADER = "{{{{Tabs:{}}}}}{{{{TOCFlat}}}}"
+        self.TEAM_TEXT = "\n\n==={{{{team|{}}}}}===\n{{{{ExtendedRoster{}{}\n}}}}"
+        self.PLAYER_TEXT = "\n|{{{{ExtendedRoster/Line{}{}\n{} }}}}"
+
+    def run(self):
+        self.get_and_process_match_data()
+        self.initialize_roster_data()
+        players_data = self.get_player_data()
+        self.process_game_data()
+        output = self.make_output(players_data)
+        self.save_page(output)
+
+    def get_and_process_match_data(self):
```

Comment:
I'm not sure this function is needed - you could probably do these 4 things in `run` and it would be fine